src/Proto3/Suite/Form/Encode/Core.hs

```diff
@@ -96,34 +98,56 @@ toLazyByteString = Encode.toLazyByteString . untypedMessageEncoder
 etaMessageEncoder :: forall a message . (a -> MessageEncoder message) -> a -> MessageEncoder message
 etaMessageEncoder = coerce (Encode.etaMessageBuilder @a)
 
--- | The octet sequence that would be emitted by
--- some 'MessageEncoder' having the same type parameter.
+-- | The octet sequence that would be emitted by some
+-- 'MessageEncoder' having the same type parameter.
 --
--- See also: 'cacheMessageEncoder'
-newtype MessageEncoding (message :: Type) =
-  UnsafeMessageEncoding { untypedMessageEncoding :: B.ByteString }
+-- See also: 'cacheMessageEncoding'
+newtype MessageEncoding (message :: Type) = UnsafeMessageEncoding B.ByteString
+  deriving newtype (Eq)
 
 type role MessageEncoding nominal
 
+instance (Message message, Show message) =>
+         Show (MessageEncoding message)
+  where
+    showsPrec d (messageEncodingToByteString -> bs) = showParen (d >= 11) $
+      case fromByteString bs of
+        Left _ ->
+          showString "Proto3.Suite.Form.Encode.unsafeByteStringToMessageEncoding " .
```

Comment:
Use `show 'unsafeByteStringToMessageEncoding` (name quoting) to ensure that this qualified name stays up to date, and to ensure we get a compile-time error should the name of `unsafeByteStringToMessageEncoding` change.

Reply:
Neat idea!  Implemented in ecd3d03a1d5c9030133c5db05eb3a5da2dbe634f .